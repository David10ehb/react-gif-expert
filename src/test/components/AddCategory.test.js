import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas al component AddCategory', () => {

    const setcategories = jest.fn(); //
    let wrapper;

    beforeEach( ()=> {
        jest.clearAllMocks();
        wrapper = shallow(
            <AddCategory setcategories={setcategories} />
        );
    })


    test('comprobando wrappe', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja de texto inputs simulando un change', () => {
        const input = wrapper.find('input');
        const value = 'mundo';
        input.simulate('change', { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('no debe de llamar la  informacion con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect( setcategories ).not.toHaveBeenCalled();
    });

    test('debe de llamar setcategories y limpiar la caja de texto', () => {
        const value = 'mundo';
        const input = wrapper.find('input');
        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect( setcategories ).toHaveBeenCalled();

        expect(input.prop('value')).toBe('');

    });



});
