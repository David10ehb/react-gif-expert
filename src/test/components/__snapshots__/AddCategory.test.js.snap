// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al component AddCategory comprobando wrappe 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    mundo
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
